.github/workflows/codeql.yml

```diff
@@ -0,0 +1,43 @@
+name: "Code Scanning - Action"
+
+on:
+  pull_request:
+
+jobs:
+  CodeQL-Build:
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
+    runs-on: ubuntu-latest
+
+    permissions:
```

Comment:
I believe this workflow implicitly requires some `read` permissions (actions and contents?) which are granted by default. Can you please make those explicit?

Reply:
Those permissions are only required for private repos. https://github.com/github/codeql-action#usage